src/catalog/src/durable/persist.rs

```diff
@@ -1154,18 +1155,18 @@ impl<T: Ord> LargeCollectionStartupCache<T> {
 struct CatalogStateInner {
     /// The [`Mode`] that this catalog was opened in.
     mode: Mode,
-    /// A cache of audit logs that is only populated during startup.
-    audit_logs: LargeCollectionStartupCache<proto::AuditLogKey>,
     /// A cache of storage usage events that is only populated during startup.
     storage_usage_events: LargeCollectionStartupCache<proto::StorageUsageKey>,
+    /// A trace of all catalog updates that can be consumed by some higher layer.
+    updates: VecDeque<memory::objects::StateUpdate>,
```

Comment:
It feels a bit like an abstraction break to use `memory::objects::StateUpdate` instead of `durable::objects::StateUpdate`. However, if we used `durable::objects::StateUpdate` then we'd end up copying some updates that end up getting ignored by the in-memory catalog.